moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1621,6 +1622,11 @@ bool PlanningScene::processAttachedCollisionObjectMsg(const moveit_msgs::msg::At
         const Eigen::Isometry3d& pose = attached_body->getGlobalPose();
         world_->addToObject(name, pose, attached_body->getShapes(), attached_body->getShapePoses());
         world_->setSubframesOfObject(name, attached_body->getSubframes());
+        const auto original_object_color = getOriginalObjectColor(name);
+        if (original_object_color.has_value())
```

Comment:
Can you add a small explanation here. Which color has the object if it does not have an original color?

Reply:
Done!